src/services/SearchResults.cpp

```diff
@@ -402,20 +403,20 @@ void SearchResults::Initialize(const SearchResults& srSource, ComparisonType nCo
     m_sSummary.append(" last known value...");
 }
 
-unsigned int SearchResults::MatchingAddressCount() noexcept
+std::size_t SearchResults::MatchingAddressCount() noexcept
 {
     if (!m_bUnfiltered)
         return m_vMatchingAddresses.size();
 
-    const unsigned int nPadding = Padding(m_nSize);
-    unsigned int nCount = 0;
+    const auto nPadding = Padding(m_nSize);
+    unsigned int nCount{};
     for (auto& block : m_vBlocks)
         nCount += block.GetSize() - nPadding;
 
     if (m_nSize == MemSize::Nibble_Lower)
         nCount *= 2;
 
-    return nCount;
+    return {nCount}; // NB: The braces are required
```

Comment:
why? just make `nCount` a `size_t`.

Reply:
The result of `block.GetSize() - nPadding` is `unsigned int`. The += with nCount as size_t would cause a narrowing error. size_t is unsigned long long in x64 based archectures. The way it was done was to satisfy both platforms. The braces disable narrowing. In Win32 it will be the same, in x64 it will be returned as a size_t without needing a narrowing conversion.

Reply:
This comment doesn't explain any of that, and as such, is not a very good comment.

I was unable to generate the error you describe. It did not appear in either the Win32 or x64 builds of the Debug or Analysis projects. There were two warnings that only appeared in the Win32 builds that should be addressed:
```
1>e:\source\raintegration\src\ra_achievementoverlay.cpp(837): warning C4018: '<': signed/unsigned mismatch
1>e:\source\raintegration\src\ra_core.cpp(1093): warning C4244: 'argument': conversion from 'std::streamsize' to 'size_t', possible loss of data
```
And one warning that only appeared in the x64 builds:
```
1>e:\source\raintegration\src\ra_dlg_acheditor.cpp(1750): warning C4267: '=': conversion from 'size_t' to 'unsigned int', possible loss of data
```